### ISSUE TITLE
Leaking state in program scope

### ISSUE THREAD

**Issue by MichalBryxi** [ember-cli/eslint-plugin-ember#446]
Consider following service:

```js
var cachedFoo = null;

export default Service.extend({
  fetchFoo(count) {
    if(!cachedFoo) {
      fetch(`./api/?results=${count}`).then(function(response) {
        cachedFoo = response;
      });
    }
  }

  getFoo() {
    return cachedFoo;
  }
});
```

This service will probably work as expected in the scope of running Ember code. 

But if there are two *tests* calling this service with different value for `count` when calling `fetchFoo`, then calls to `getFoo` will always return the response of the first test.

**Comment by MichalBryxi** [comment]
This just bit us in our test code, so I thought an eslint rule might be a good idea.

I know that just enforcing `consts` is not bullet-proof as the _content_ of an object can be still mutated, but forcing user to use `const` instead of `var` or `let` might give them a hint that something might be wrong.

- This is my first eslint rule, so not really sure what I might have missed.
- Also not sure whether there might be files that actually can have global mutable variables without compromising the tests.

**Comment by rajasegar** [comment]
@MichalBryxi FYI
https://eslint.org/docs/rules/prefer-const
https://eslint.org/docs/rules/no-var
